### ISSUE TITLE
Graceful Master Takeover is not working for me

### ISSUE THREAD

**Issue by robertmyhill3** [openark/orchestrator#514]
Hi there!

I've just started using Orchestrator in my test environment, hoping to roll it out to production as soon as I get it properly configured.  Great stuff!

I've run through several scenarios, and they are (almost) all working as expected.  The one scenario I'm having difficulties with is the graceful takeover.

I have a very simple configuration:  One master and one slave, doing replication using GTID.

With the master and slave both up and running and replicating successfully, I use the Web UI to drag-and-drop the Slave to become the Master.  I drag the Slave on top of the Master, until I see "Promote as Master".  Then I see the Danger Zone message, with graceful-master-takeover.  I click "OK".

If I understand things correctly, the Slave should now become the Master, and vice versa.  Assuming my understanding is correct (please let me know if I'm wrong), then this is where the problems start.

On the Web UI, the Master remains the Master, and the Slave has turned red, indicating some error has happened.

Below is the log output, starting at the point when I click "OK" from the Danger Zone message.

Any thoughts on what the problem might be?

Best,
Robert

+++++++++++++++++++++++++++++++++++++++++++++++++++

2018-05-22 15:36:49 DEBUG outdated keys: [ubuntu:3306 ubuntu2:3306]
2018-05-22 15:36:49 DEBUG Discovered host: ubuntu:3306, master: ubuntu2:3306, version: 5.7.17-13-log in 0.013s (Backend: 0.004s, Instance: 0.010s)
2018-05-22 15:36:49 ERROR ReadTopologyInstance(ubuntu2:3306) show slave hosts: ReadTopologyInstance(ubuntu2:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-22 15:36:49 DEBUG Discovered host: ubuntu2:3306, master: :0, version: 5.7.17-13-log in 0.019s (Backend: 0.004s, Instance: 0.017s)
[martini] Started GET /api/graceful-master-takeover/ubuntu2/3306 for 127.0.0.1:52020
2018-05-22 15:36:50 ERROR ReadTopologyInstance(ubuntu2:3306) show slave hosts: ReadTopologyInstance(ubuntu2:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-22 15:36:50 INFO Will demote ubuntu2:3306 and promote ubuntu:3306 instead
[mysql] 2018/05/22 15:36:50 packets.go:36: unexpected EOF
2018-05-22 15:36:50 INFO topology_recovery: Running 1 PreGracefulTakeoverProcesses hooks
2018-05-22 15:36:50 INFO topology_recovery: Running PreGracefulTakeoverProcesses hook 1 of 1: echo 'Planned takeover about to take place on ubuntu2:3306. Master will switch to read_only' >> /tmp/recovery.log
2018-05-22 15:36:50 INFO CommandRun(echo 'Planned takeover about to take place on ubuntu2:3306. Master will switch to read_only' >> /tmp/recovery.log,[])
2018-05-22 15:36:50 INFO CommandRun/running: bash /tmp/orchestrator-process-cmd-044302037
2018-05-22 15:36:50 INFO CommandRun successful. exit status 0
2018-05-22 15:36:50 INFO topology_recovery: Completed PreGracefulTakeoverProcesses hook 1 of 1 in 6.4657ms
2018-05-22 15:36:50 INFO Completed PreGracefulTakeoverProcesses hook 1 of 1 in 6.4657ms
2018-05-22 15:36:50 INFO topology_recovery: done running PreGracefulTakeoverProcesses hooks
[mysql] 2018/05/22 15:36:50 connection.go:311: invalid connection
2018-05-22 15:36:50 INFO Stopped slave on ubuntu:3306, Self:log-bin-local.000001:154, Exec:log-bin-local.000001:154
2018-05-22 15:36:50 INFO Will set ubuntu2:3306 as read_only
2018-05-22 15:36:50 ERROR ReadTopologyInstance(ubuntu2:3306) show slave hosts: ReadTopologyInstance(ubuntu2:3306) 'show slave hosts' returned row with <host,port>: <,3306>
[mysql] 2018/05/22 15:36:50 connection.go:311: invalid connection
2018-05-22 15:36:50 ERROR ReadTopologyInstance(ubuntu2:3306) show slave hosts: ReadTopologyInstance(ubuntu2:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-22 15:36:50 INFO instance ubuntu2:3306 read_only: true
2018-05-22 15:36:50 INFO auditType:read-only instance:ubuntu2:3306 cluster:ubuntu2:3306 message:set as true
2018-05-22 15:36:50 INFO Will advance ubuntu:3306 to master coordinates log-bin-local.000001:154
2018-05-22 15:36:50 INFO Will start slave on ubuntu:3306 until coordinates: log-bin-local.000001:154
2018-05-22 15:36:50 INFO Stopped slave on ubuntu:3306, Self:log-bin-local.000001:154, Exec:log-bin-local.000001:154
2018-05-22 15:36:50 INFO executeCheckAndRecoverFunction: proceeding with DeadMaster detection on ubuntu2:3306; isActionable?: true; skipProcesses: false
2018-05-22 15:36:50 INFO topology_recovery: detected DeadMaster failure on ubuntu2:3306
2018-05-22 15:36:50 INFO topology_recovery: Running 1 OnFailureDetectionProcesses hooks
2018-05-22 15:36:50 INFO topology_recovery: Running OnFailureDetectionProcesses hook 1 of 1: echo 'Detected DeadMaster on ubuntu2:3306. Affected replicas: 0' >> /tmp/recovery.log
2018-05-22 15:36:50 INFO CommandRun(echo 'Detected DeadMaster on ubuntu2:3306. Affected replicas: 0' >> /tmp/recovery.log,[])
2018-05-22 15:36:50 INFO CommandRun/running: bash /tmp/orchestrator-process-cmd-316604464
2018-05-22 15:36:50 INFO CommandRun successful. exit status 0
2018-05-22 15:36:50 INFO topology_recovery: Completed OnFailureDetectionProcesses hook 1 of 1 in 2.163731ms
2018-05-22 15:36:50 INFO Completed OnFailureDetectionProcesses hook 1 of 1 in 2.163731ms
2018-05-22 15:36:50 INFO topology_recovery: done running OnFailureDetectionProcesses hooks
2018-05-22 15:36:50 INFO executeCheckAndRecoverFunction: proceeding with DeadMaster recovery on ubuntu2:3306; isRecoverable?: true; skipProcesses: false
2018-05-22 15:36:50 INFO topology_recovery: will handle DeadMaster event on ubuntu2:3306
2018-05-22 15:36:50 INFO auditType:recover-dead-master instance:ubuntu2:3306 cluster:ubuntu2:3306 message:problem found; will recover
2018-05-22 15:36:50 INFO topology_recovery: Running 1 PreFailoverProcesses hooks
2018-05-22 15:36:50 INFO topology_recovery: Running PreFailoverProcesses hook 1 of 1: echo 'Will recover from DeadMaster on ubuntu2:3306' >> /tmp/recovery.log
2018-05-22 15:36:50 INFO CommandRun(echo 'Will recover from DeadMaster on ubuntu2:3306' >> /tmp/recovery.log,[])
2018-05-22 15:36:50 INFO CommandRun/running: bash /tmp/orchestrator-process-cmd-111657935
2018-05-22 15:36:50 INFO CommandRun successful. exit status 0
2018-05-22 15:36:50 INFO topology_recovery: Completed PreFailoverProcesses hook 1 of 1 in 2.186568ms
2018-05-22 15:36:50 INFO Completed PreFailoverProcesses hook 1 of 1 in 2.186568ms
2018-05-22 15:36:50 INFO topology_recovery: done running PreFailoverProcesses hooks
2018-05-22 15:36:50 INFO topology_recovery: RecoverDeadMaster: will recover ubuntu2:3306
2018-05-22 15:36:50 INFO topology_recovery: RecoverDeadMaster: masterRecoveryType=MasterRecoveryPseudoGTID
2018-05-22 15:36:50 INFO topology_recovery: RecoverDeadMaster: regrouping replicas via Pseudo-GTID
2018-05-22 15:36:50 DEBUG RegroupReplicasIncludingSubReplicasOfBinlogServers: starting on replicas of ubuntu2:3306
2018-05-22 15:36:50 DEBUG RegroupReplicasIncludingSubReplicasOfBinlogServers: no binlog server replicates from ubuntu2:3306
2018-05-22 15:36:50 DEBUG Stopping 1 replicas via StopReplicationNicely
2018-05-22 15:36:50 INFO Stopped slave nicely on ubuntu:3306, Self:log-bin-local.000001:154, Exec:log-bin-local.000001:154
2018-05-22 15:36:51 INFO Stopped slave on ubuntu:3306, Self:log-bin-local.000001:154, Exec:log-bin-local.000001:154
2018-05-22 15:36:51 DEBUG - sorted replica: ubuntu:3306 log-bin-local.000001:154
2018-05-22 15:36:51 DEBUG GetCandidateReplica: candidate: ubuntu:3306, ahead: 0, equal: 0, late: 0, break: 0
2018-05-22 15:36:51 INFO auditType:begin-downtime instance:ubuntu2:3306 cluster:ubuntu2:3306 message:owner: ubuntu3, reason: lost-in-recovery
2018-05-22 15:36:51 INFO topology_recovery: RecoverDeadMaster: 0 postponed functions
2018-05-22 15:36:51 INFO topology_recovery: checking if should replace promoted replica with a better candidate
2018-05-22 15:36:51 INFO topology_recovery: + sanity check: found our very own server to promote; doing nothing
2018-05-22 15:36:51 INFO topology_recovery: promoted instance ubuntu:3306 requires no further action
2018-05-22 15:36:51 INFO topology_recovery: promoted replica: ubuntu:3306
2018-05-22 15:36:51 INFO auditType:recover-dead-master instance:ubuntu2:3306 cluster:ubuntu2:3306 message:promoted replica: ubuntu:3306
2018-05-22 15:36:51 INFO topology_recovery: RecoverDeadMaster: successfully promoted ubuntu:3306
2018-05-22 15:36:51 INFO topology_recovery: - RecoverDeadMaster: promoted server coordinates: log-bin-local.000001:154
2018-05-22 15:36:51 INFO topology_recovery: - RecoverDeadMaster: will apply MySQL changes to promoted master
2018-05-22 15:36:51 INFO Will reset replica on ubuntu:3306
2018-05-22 15:36:51 INFO auditType:begin-maintenance instance:ubuntu:3306 cluster:ubuntu2:3306 message:maintenanceToken: 36, owner: ubuntu3, reason: reset replica
2018-05-22 15:36:51 INFO Stopped slave on ubuntu:3306, Self:log-bin-local.000001:154, Exec:log-bin-local.000001:154
2018-05-22 15:36:51 INFO Reset slave ubuntu:3306
2018-05-22 15:36:51 INFO auditType:reset-slave instance:ubuntu:3306 cluster:ubuntu2:3306 message:ubuntu:3306 replication reset
2018-05-22 15:36:51 INFO auditType:end-maintenance instance:ubuntu:3306 cluster:ubuntu2:3306 message:maintenanceToken: 36
2018-05-22 15:36:51 INFO instance ubuntu:3306 read_only: false
2018-05-22 15:36:51 INFO auditType:read-only instance:ubuntu:3306 cluster:ubuntu2:3306 message:set as false
2018-05-22 15:36:51 INFO topology_recovery: - RecoverDeadMaster: have applied read-only=0, RESET SLAVE ALL on promtoed master
2018-05-22 15:36:51 INFO topology_recovery: Writing KV [mysql/master/ashburn_cluster:ubuntu:3306 mysql/master/ashburn_cluster/hostname:ubuntu mysql/master/ashburn_cluster/port:3306 mysql/master/ashburn_cluster/ipv4:192.168.0.143 mysql/master/ashburn_cluster/ipv6:]
2018-05-22 15:36:51 INFO topology_recovery: Running 1 PostMasterFailoverProcesses hooks
2018-05-22 15:36:51 INFO topology_recovery: Running PostMasterFailoverProcesses hook 1 of 1: echo 'Recovered from DeadMaster on ubuntu2:3306. Failed: ubuntu2:3306; Promoted: ubuntu:3306' >> /tmp/recovery.log
2018-05-22 15:36:51 INFO CommandRun(echo 'Recovered from DeadMaster on ubuntu2:3306. Failed: ubuntu2:3306; Promoted: ubuntu:3306' >> /tmp/recovery.log,[])
2018-05-22 15:36:51 INFO CommandRun/running: bash /tmp/orchestrator-process-cmd-692394210
2018-05-22 15:36:51 INFO CommandRun successful. exit status 0
2018-05-22 15:36:51 INFO topology_recovery: Completed PostMasterFailoverProcesses hook 1 of 1 in 7.748975ms
2018-05-22 15:36:51 INFO Completed PostMasterFailoverProcesses hook 1 of 1 in 7.748975ms
2018-05-22 15:36:51 INFO topology_recovery: done running PostMasterFailoverProcesses hooks
2018-05-22 15:36:51 INFO topology_recovery: - RecoverDeadMaster: updating cluster_alias: ubuntu2:3306 -> ubuntu:3306
2018-05-22 15:36:51 ERROR ReadTopologyInstance(ubuntu2:3306) show slave hosts: ReadTopologyInstance(ubuntu2:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-22 15:36:51 INFO Topology recovery: {"Id":44,"UID":"1527017810873342303:4879591e328e2b630dc4577ee68199379410e878517a6fb171288c7829c269c0","AnalysisEntry":{"AnalyzedInstanceKey":{"Hostname":"ubuntu2","Port":3306},"AnalyzedInstanceMasterKey":{"Hostname":"","Port":0},"ClusterDetails":{"ClusterName":"ubuntu2:3306","ClusterAlias":"ashburn_cluster","ClusterDomain":"","CountInstances":2,"HeuristicLag":0,"HasAutomatedMasterRecovery":true,"HasAutomatedIntermediateMasterRecovery":false},"AnalyzedInstanceDataCenter":"","AnalyzedInstancePhysicalEnvironment":"","IsMaster":false,"IsCoMaster":false,"LastCheckValid":false,"LastCheckPartialSuccess":false,"CountReplicas":0,"CountValidReplicas":0,"CountValidReplicatingReplicas":0,"CountReplicasFailingToConnectToMaster":0,"CountStaleReplicas":0,"CountDowntimedReplicas":0,"ReplicationDepth":0,"SlaveHosts":[],"IsFailingToConnectToMaster":false,"Analysis":"DeadMaster","Description":"","StructureAnalysis":null,"IsDowntimed":false,"IsReplicasDowntimed":false,"DowntimeEndTimestamp":"","DowntimeRemainingSeconds":0,"IsBinlogServer":false,"PseudoGTIDImmediateTopology":false,"OracleGTIDImmediateTopology":false,"MariaDBGTIDImmediateTopology":false,"BinlogServerImmediateTopology":false,"CountStatementBasedLoggingReplicas":0,"CountMixedBasedLoggingReplicas":0,"CountRowBasedLoggingReplicas":0,"CountDistinctMajorVersionsLoggingReplicas":0,"IsActionableRecovery":true,"ProcessingNodeHostname":"","ProcessingNodeToken":"","CountAdditionalAgreeingNodes":0,"StartActivePeriod":"","SkippableDueToDowntime":false,"GTIDMode":"","MinReplicaGTIDMode":"","MaxReplicaGTIDMode":"","CommandHint":"graceful-master-takeover"},"SuccessorKey":{"Hostname":"ubuntu","Port":3306},"SuccessorAlias":"","IsActive":false,"IsSuccessful":true,"LostReplicas":[],"ParticipatingInstanceKeys":[],"AllErrors":["PseudoGTIDPattern not configured; cannot use Pseudo-GTID"],"RecoveryStartTimestamp":"","RecoveryEndTimestamp":"","ProcessingNodeHostname":"","ProcessingNodeToken":"","Acknowledged":false,"AcknowledgedAt":"","AcknowledgedBy":"","AcknowledgedComment":"","LastDetectionId":0,"RelatedRecoveryId":0,"RecoveryType":"MasterRecoveryPseudoGTID"}
2018-05-22 15:36:51 INFO topology_recovery: Running 1 PostFailoverProcesses hooks
2018-05-22 15:36:51 INFO topology_recovery: Running PostFailoverProcesses hook 1 of 1: echo '(for all types) Recovered from DeadMaster on ubuntu2:3306. Failed: ubuntu2:3306; Successor: ubuntu:3306' >> /tmp/recovery.log
2018-05-22 15:36:51 INFO CommandRun(echo '(for all types) Recovered from DeadMaster on ubuntu2:3306. Failed: ubuntu2:3306; Successor: ubuntu:3306' >> /tmp/recovery.log,[])
2018-05-22 15:36:51 INFO CommandRun/running: bash /tmp/orchestrator-process-cmd-091563737
2018-05-22 15:36:51 INFO CommandRun successful. exit status 0
2018-05-22 15:36:51 INFO topology_recovery: Completed PostFailoverProcesses hook 1 of 1 in 2.969143ms
2018-05-22 15:36:51 INFO Completed PostFailoverProcesses hook 1 of 1 in 2.969143ms
2018-05-22 15:36:51 INFO topology_recovery: done running PostFailoverProcesses hooks
2018-05-22 15:36:51 INFO topology_recovery: Waiting for 0 postponed functions
2018-05-22 15:36:51 DEBUG PostponedFunctionsContainer: waiting on 0 postponed functions
2018-05-22 15:36:51 DEBUG PostponedFunctionsContainer: done waiting
2018-05-22 15:36:51 INFO topology_recovery: Executed 0 postponed functions
2018-05-22 15:36:51 ERROR ReadTopologyInstance(ubuntu2:3306) show slave hosts: ReadTopologyInstance(ubuntu2:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-22 15:36:51 ERROR ReadTopologyInstance(ubuntu2:3306) show slave hosts: ReadTopologyInstance(ubuntu2:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-22 15:36:51 DEBUG ChangeMasterTo: will attempt changing master on ubuntu2:3306 to ubuntu:3306, log-bin-local.000001:154
2018-05-22 15:36:51 INFO instance ubuntu2:3306 read_only: true
2018-05-22 15:36:51 INFO auditType:read-only instance:ubuntu2:3306 cluster:ubuntu2:3306 message:set as true
2018-05-22 15:36:51 INFO ChangeMasterTo: Changed master on ubuntu2:3306 to: ubuntu:3306, log-bin-local.000001:154. GTID: false
2018-05-22 15:36:51 ERROR ReadTopologyInstance(ubuntu2:3306) show slave hosts: ReadTopologyInstance(ubuntu2:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-22 15:36:51 INFO topology_recovery: Running 1 PostGracefulTakeoverProcesses hooks
2018-05-22 15:36:51 INFO topology_recovery: Running PostGracefulTakeoverProcesses hook 1 of 1: echo 'Planned takeover complete' >> /tmp/recovery.log
2018-05-22 15:36:51 INFO CommandRun(echo 'Planned takeover complete' >> /tmp/recovery.log,[])
2018-05-22 15:36:51 INFO CommandRun/running: bash /tmp/orchestrator-process-cmd-142302820
2018-05-22 15:36:51 INFO CommandRun successful. exit status 0
2018-05-22 15:36:51 INFO topology_recovery: Completed PostGracefulTakeoverProcesses hook 1 of 1 in 2.17535ms
2018-05-22 15:36:51 INFO Completed PostGracefulTakeoverProcesses hook 1 of 1 in 2.17535ms
2018-05-22 15:36:51 INFO topology_recovery: done running PostGracefulTakeoverProcesses hooks
[martini] Completed 200 OK in 803.054951ms
[martini] Started GET /web/cluster/alias/ashburn_cluster for 127.0.0.1:52020
[martini] Completed 200 OK in 3.461061ms
[martini] Started GET /api/clusters-info for 127.0.0.1:52020
[martini] Completed 200 OK in 1.908072ms
[martini] Started GET /api/cluster/ubuntu:3306 for 127.0.0.1:52020
[martini] Completed 200 OK in 3.47226ms
[martini] Started GET /api/cluster-info/ubuntu:3306 for 127.0.0.1:52018
[martini] Started GET /api/recently-active-cluster-recovery/ubuntu:3306 for 127.0.0.1:52022
[martini] Started GET /api/active-cluster-recovery/ubuntu:3306 for 127.0.0.1:52024
[martini] Completed 200 OK in 985.403µs
[martini] Completed 200 OK in 7.161234ms
[martini] Completed 200 OK in 7.852909ms
[martini] Started GET /api/problems/ubuntu:3306 for 127.0.0.1:52018
[martini] Started GET /api/blocked-recoveries/cluster/ubuntu:3306 for 127.0.0.1:52020
[martini] Completed 200 OK in 2.200838ms
[martini] Completed 200 OK in 3.007532ms
[martini] Started GET /api/replication-analysis/ubuntu:3306 for 127.0.0.1:52024
[martini] Started GET /api/audit-recovery/alias/ashburn_cluster for 127.0.0.1:52020
[martini] Started GET /api/audit-failure-detection/alias/ashburn_cluster for 127.0.0.1:52018
[martini] Completed 200 OK in 6.239722ms
[martini] Completed 200 OK in 19.133034ms
[martini] Completed 200 OK in 19.371882ms
[martini] Started GET /api/maintenance for 127.0.0.1:52024
[martini] Completed 200 OK in 2.467907ms
[martini] Started GET /api/maintenance for 127.0.0.1:52020
[martini] Completed 200 OK in 5.279833ms
[martini] Started GET /api/recently-active-instance-recovery/ubuntu/3306 for 127.0.0.1:52020
[martini] Completed 200 OK in 6.351461ms
2018-05-22 15:36:52 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-22 15:36:53 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-22 15:36:54 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-22 15:36:55 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-22 15:36:56 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-22 15:36:57 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-22 15:36:57 DEBUG outdated keys: [ubuntu:3306 ubuntu2:3306]
2018-05-22 15:36:57 ERROR ReadTopologyInstance(ubuntu2:3306) show slave hosts: ReadTopologyInstance(ubuntu2:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-22 15:36:57 DEBUG Discovered host: ubuntu:3306, master: :0, version: 5.7.17-13-log in 0.015s (Backend: 0.004s, Instance: 0.013s)
2018-05-22 15:36:57 DEBUG Discovered host: ubuntu2:3306, master: ubuntu:3306, version: 5.7.17-13-log in 0.018s (Backend: 0.008s, Instance: 0.012s)
2018-05-22 15:36:58 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-22 15:36:59 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-22 15:37:00 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-22 15:37:01 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-22 15:37:02 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-22 15:37:03 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2

**Comment by shlomi-noach** [comment]
A few things. 

- I see you got: `"RecoveryType":"MasterRecoveryPseudoGTID"` as opposed to `MasterRecoveryGTID`. Just yesterday I released [3.0.11](https://github.com/github/orchestrator/releases/tag/v3.0.11) with a fix to GTID graceful takeovers which may address this. Please first make sure to use that version.

- Make sure your replica uses `AUTO_POSITION: 1` (see `show slave status\G`) or else `orchestrator` will not actually _use_ GTID.

- I do see you have `2018-05-22 15:36:51 INFO topology_recovery: - RecoverDeadMaster: have applied read-only=0, RESET SLAVE ALL on promtoed master` (typo included). This is good. This also means the new master should _not_ show up as a replica. Can you please verify: `show slave status` on promoted server? Does it still replicate? It is still `read_only`?

**Comment by robertmyhill3** [comment]
Thanks for your quick reply.

I ran the same test using 3.0.11, and the results are a little bit different.  The difference between 3.0.10 and 3.0.11 behavior is:

With 3.0.11...
- The first time I try to do a graceful takeover, I get a "Invalid Connection" message on the web UI.
- The second time I try to do a graceful takeover it almost works. The only think is that I have to issue a Start Slave command from the web UI, and then the graceful takeover is successful.

Question: Is the intent with a graceful takeover that it is completely automatic, or is it by design that a Start Slave on the new slave is required to complete the takeover?

Here are the details:

- I upgraded to 3.0.11
- I re-ran the test twice with identical results
- I confirmed "RecoveryType":"MasterRecoveryGTID" (see log below)
- I confirmed that the replica uses AUTO_POSITION: 1.
- SHOW SLAVE STATUS on the promoted master has an empty result set
- read_only and super_read_only are 0 on the promoted master
- When I tried a graceful takeover I got an Invalid Connection error (which you can see in the log below.) Then I retried the graceful takeover, and the takeover almost completed - I just needed to issue a Start Slave on the new slave.
- Below is the log from today's 3.0.11 test
- After the log, at the bottom of this message, there is a screen shot of the Web UI status of the situation.  "ubuntu2" is the new master, which is correct.  The problem is that "ubuntu" has not been full setup to be the new slave.  When I clicked on Start Slave, the failover completed.

Thanks again.
Robert

++++++++++++++++++++++++++++++++++++++++++++++

[martini] Completed 200 OK in 2.54878ms
2018-05-23 13:48:32 DEBUG outdated keys: [ubuntu2:3306]
2018-05-23 13:48:32 DEBUG Discovered host: ubuntu2:3306, master: ubuntu:3306, version: 5.7.17-13-log in 0.011s (Backend: 0.003s, Instance: 0.009s)
2018-05-23 13:48:34 DEBUG outdated keys: [ubuntu:3306]
2018-05-23 13:48:34 ERROR ReadTopologyInstance(ubuntu:3306) show slave hosts: ReadTopologyInstance(ubuntu:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-23 13:48:34 DEBUG Discovered host: ubuntu:3306, master: :0, version: 5.7.17-13-log in 0.110s (Backend: 0.021s, Instance: 0.091s)
2018-05-23 13:48:38 DEBUG outdated keys: [ubuntu2:3306]
2018-05-23 13:48:38 DEBUG Discovered host: ubuntu2:3306, master: ubuntu:3306, version: 5.7.17-13-log in 0.015s (Backend: 0.006s, Instance: 0.012s)
2018-05-23 13:48:40 DEBUG outdated keys: [ubuntu:3306]
2018-05-23 13:48:40 ERROR ReadTopologyInstance(ubuntu:3306) show slave hosts: ReadTopologyInstance(ubuntu:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-23 13:48:40 DEBUG Discovered host: ubuntu:3306, master: :0, version: 5.7.17-13-log in 0.040s (Backend: 0.029s, Instance: 0.016s)
2018-05-23 13:48:44 DEBUG outdated keys: [ubuntu2:3306]
2018-05-23 13:48:44 DEBUG Discovered host: ubuntu2:3306, master: ubuntu:3306, version: 5.7.17-13-log in 0.019s (Backend: 0.004s, Instance: 0.016s)
[martini] Started GET /api/graceful-master-takeover/ubuntu/3306/ubuntu2/3306 for 127.0.0.1:39674
2018-05-23 13:48:45 INFO GracefulMasterTakeover: designated master instructed to be ubuntu2:3306
2018-05-23 13:48:45 ERROR ReadTopologyInstance(ubuntu:3306) show slave hosts: ReadTopologyInstance(ubuntu:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-23 13:48:45 INFO GracefulMasterTakeover: Will demote ubuntu:3306 and promote ubuntu2:3306 instead
[mysql] 2018/05/23 13:48:45 packets.go:36: unexpected EOF
2018-05-23 13:48:45 INFO topology_recovery: Running 1 PreGracefulTakeoverProcesses hooks
2018-05-23 13:48:45 INFO topology_recovery: Running PreGracefulTakeoverProcesses hook 1 of 1: echo 'Planned takeover about to take place on ubuntu:3306. Master will switch to read_only' >> /tmp/recovery.log
2018-05-23 13:48:45 INFO CommandRun(echo 'Planned takeover about to take place on ubuntu:3306. Master will switch to read_only' >> /tmp/recovery.log,[])
2018-05-23 13:48:45 INFO CommandRun/running: bash /tmp/orchestrator-process-cmd-189453755
2018-05-23 13:48:45 INFO CommandRun successful. exit status 0
2018-05-23 13:48:45 INFO topology_recovery: Completed PreGracefulTakeoverProcesses hook 1 of 1 in 1.506289ms
2018-05-23 13:48:45 INFO Completed PreGracefulTakeoverProcesses hook 1 of 1 in 1.506289ms
2018-05-23 13:48:45 INFO topology_recovery: done running PreGracefulTakeoverProcesses hooks
[mysql] 2018/05/23 13:48:45 connection.go:311: invalid connection
2018-05-23 13:48:45 INFO Stopped slave on ubuntu2:3306, Self:log-bin-local.000001:154, Exec:log-bin-local.000001:154
2018-05-23 13:48:45 INFO Will set ubuntu:3306 as read_only
2018-05-23 13:48:45 ERROR ReadTopologyInstance(ubuntu:3306) show slave hosts: ReadTopologyInstance(ubuntu:3306) 'show slave hosts' returned row with <host,port>: <,3306>
[mysql] 2018/05/23 13:48:45 packets.go:36: unexpected EOF
2018-05-23 13:48:45 ERROR invalid connection
2018-05-23 13:48:45 ERROR invalid connection
[martini] Completed 500 Internal Server Error in 78.409617ms
2018-05-23 13:48:45 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:48:46 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:48:47 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:48:48 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:48:49 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:48:50 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:48:51 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:48:51 DEBUG outdated keys: [ubuntu:3306 ubuntu2:3306]
2018-05-23 13:48:51 ERROR ReadTopologyInstance(ubuntu:3306) show slave hosts: ReadTopologyInstance(ubuntu:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-23 13:48:51 DEBUG Discovered host: ubuntu:3306, master: :0, version: 5.7.17-13-log in 0.014s (Backend: 0.004s, Instance: 0.011s)
2018-05-23 13:48:51 DEBUG Discovered host: ubuntu2:3306, master: ubuntu:3306, version: 5.7.17-13-log in 0.014s (Backend: 0.005s, Instance: 0.012s)
2018-05-23 13:48:52 ERROR FlushNontrivialResolveCacheToDatabase() called, but HostnameResolveMethod is default
2018-05-23 13:48:52 INFO auditType:resolve-unknown-masters instance::0 cluster: message:Num resolved hostnames: 0
2018-05-23 13:48:52 INFO auditType:inject-unseen-masters instance::0 cluster: message:Operations: 0
2018-05-23 13:48:52 INFO auditType:forget-unseen instance::0 cluster: message:Forgotten instances: 0
2018-05-23 13:48:52 INFO auditType:review-unseen-instances instance::0 cluster: message:Operations: 0
2018-05-23 13:48:52 INFO auditType:forget-unseen-differently-resolved instance::0 cluster: message:Forgotten instances: 0
2018-05-23 13:48:52 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:48:53 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:48:54 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:48:55 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:48:56 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:48:57 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:48:57 DEBUG outdated keys: [ubuntu:3306 ubuntu2:3306]
2018-05-23 13:48:57 ERROR ReadTopologyInstance(ubuntu:3306) show slave hosts: ReadTopologyInstance(ubuntu:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-23 13:48:57 DEBUG Discovered host: ubuntu2:3306, master: ubuntu:3306, version: 5.7.17-13-log in 0.014s (Backend: 0.005s, Instance: 0.012s)
2018-05-23 13:48:57 DEBUG Discovered host: ubuntu:3306, master: :0, version: 5.7.17-13-log in 0.016s (Backend: 0.004s, Instance: 0.014s)
2018-05-23 13:48:58 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:48:59 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:49:00 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:49:01 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:49:02 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:49:03 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:49:03 DEBUG outdated keys: [ubuntu:3306 ubuntu2:3306]
2018-05-23 13:49:03 ERROR ReadTopologyInstance(ubuntu:3306) show slave hosts: ReadTopologyInstance(ubuntu:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-23 13:49:03 DEBUG Discovered host: ubuntu2:3306, master: ubuntu:3306, version: 5.7.17-13-log in 0.017s (Backend: 0.004s, Instance: 0.014s)
2018-05-23 13:49:03 DEBUG Discovered host: ubuntu:3306, master: :0, version: 5.7.17-13-log in 0.017s (Backend: 0.006s, Instance: 0.014s)
2018-05-23 13:49:04 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:49:05 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:49:06 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:49:07 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:49:08 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:49:09 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:49:09 DEBUG outdated keys: [ubuntu:3306 ubuntu2:3306]
2018-05-23 13:49:09 ERROR ReadTopologyInstance(ubuntu:3306) show slave hosts: ReadTopologyInstance(ubuntu:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-23 13:49:09 DEBUG Discovered host: ubuntu:3306, master: :0, version: 5.7.17-13-log in 0.027s (Backend: 0.008s, Instance: 0.021s)
2018-05-23 13:49:09 DEBUG Discovered host: ubuntu2:3306, master: ubuntu:3306, version: 5.7.17-13-log in 0.033s (Backend: 0.008s, Instance: 0.029s)
2018-05-23 13:49:10 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
2018-05-23 13:49:11 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu:3306
[martini] Started GET /api/graceful-master-takeover/ubuntu/3306/ubuntu2/3306 for 127.0.0.1:39674
2018-05-23 13:49:11 INFO GracefulMasterTakeover: designated master instructed to be ubuntu2:3306
2018-05-23 13:49:11 ERROR ReadTopologyInstance(ubuntu:3306) show slave hosts: ReadTopologyInstance(ubuntu:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-23 13:49:11 INFO GracefulMasterTakeover: Will demote ubuntu:3306 and promote ubuntu2:3306 instead
2018-05-23 13:49:11 INFO topology_recovery: Running 1 PreGracefulTakeoverProcesses hooks
2018-05-23 13:49:11 INFO topology_recovery: Running PreGracefulTakeoverProcesses hook 1 of 1: echo 'Planned takeover about to take place on ubuntu:3306. Master will switch to read_only' >> /tmp/recovery.log
2018-05-23 13:49:11 INFO CommandRun(echo 'Planned takeover about to take place on ubuntu:3306. Master will switch to read_only' >> /tmp/recovery.log,[])
2018-05-23 13:49:11 INFO CommandRun/running: bash /tmp/orchestrator-process-cmd-889739230
2018-05-23 13:49:11 INFO CommandRun successful. exit status 0
2018-05-23 13:49:11 INFO topology_recovery: Completed PreGracefulTakeoverProcesses hook 1 of 1 in 3.285989ms
2018-05-23 13:49:11 INFO Completed PreGracefulTakeoverProcesses hook 1 of 1 in 3.285989ms
2018-05-23 13:49:11 INFO topology_recovery: done running PreGracefulTakeoverProcesses hooks
2018-05-23 13:49:11 INFO Stopped slave on ubuntu2:3306, Self:log-bin-local.000001:154, Exec:log-bin-local.000001:154
2018-05-23 13:49:11 INFO Will set ubuntu:3306 as read_only
2018-05-23 13:49:11 ERROR ReadTopologyInstance(ubuntu:3306) show slave hosts: ReadTopologyInstance(ubuntu:3306) 'show slave hosts' returned row with <host,port>: <,3306>
[mysql] 2018/05/23 13:49:11 connection.go:311: invalid connection
2018-05-23 13:49:11 ERROR ReadTopologyInstance(ubuntu:3306) show slave hosts: ReadTopologyInstance(ubuntu:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-23 13:49:11 INFO instance ubuntu:3306 read_only: true
2018-05-23 13:49:11 INFO auditType:read-only instance:ubuntu:3306 cluster:ubuntu:3306 message:set as true
2018-05-23 13:49:11 INFO Will advance ubuntu2:3306 to master coordinates log-bin-local.000001:154
2018-05-23 13:49:11 INFO Will start slave on ubuntu2:3306 until coordinates: log-bin-local.000001:154
2018-05-23 13:49:11 INFO Stopped slave on ubuntu2:3306, Self:log-bin-local.000001:154, Exec:log-bin-local.000001:154
2018-05-23 13:49:11 INFO executeCheckAndRecoverFunction: proceeding with DeadMaster detection on ubuntu:3306; isActionable?: true; skipProcesses: false
2018-05-23 13:49:11 INFO topology_recovery: detected DeadMaster failure on ubuntu:3306
2018-05-23 13:49:11 INFO topology_recovery: Running 1 OnFailureDetectionProcesses hooks
2018-05-23 13:49:11 INFO topology_recovery: Running OnFailureDetectionProcesses hook 1 of 1: echo 'Detected DeadMaster on ubuntu:3306. Affected replicas: 1' >> /tmp/recovery.log
2018-05-23 13:49:11 INFO CommandRun(echo 'Detected DeadMaster on ubuntu:3306. Affected replicas: 1' >> /tmp/recovery.log,[])
2018-05-23 13:49:11 INFO CommandRun/running: bash /tmp/orchestrator-process-cmd-582713253
2018-05-23 13:49:11 INFO CommandRun successful. exit status 0
2018-05-23 13:49:11 INFO topology_recovery: Completed OnFailureDetectionProcesses hook 1 of 1 in 2.938916ms
2018-05-23 13:49:11 INFO Completed OnFailureDetectionProcesses hook 1 of 1 in 2.938916ms
2018-05-23 13:49:11 INFO topology_recovery: done running OnFailureDetectionProcesses hooks
2018-05-23 13:49:11 INFO executeCheckAndRecoverFunction: proceeding with DeadMaster recovery on ubuntu:3306; isRecoverable?: true; skipProcesses: false
2018-05-23 13:49:11 INFO topology_recovery: will handle DeadMaster event on ubuntu:3306
2018-05-23 13:49:11 INFO auditType:recover-dead-master instance:ubuntu:3306 cluster:ubuntu:3306 message:problem found; will recover
2018-05-23 13:49:11 INFO topology_recovery: Running 1 PreFailoverProcesses hooks
2018-05-23 13:49:11 INFO topology_recovery: Running PreFailoverProcesses hook 1 of 1: echo 'Will recover from DeadMaster on ubuntu:3306' >> /tmp/recovery.log
2018-05-23 13:49:11 INFO CommandRun(echo 'Will recover from DeadMaster on ubuntu:3306' >> /tmp/recovery.log,[])
2018-05-23 13:49:11 INFO CommandRun/running: bash /tmp/orchestrator-process-cmd-945498304
2018-05-23 13:49:11 INFO CommandRun successful. exit status 0
2018-05-23 13:49:11 INFO topology_recovery: Completed PreFailoverProcesses hook 1 of 1 in 2.708585ms
2018-05-23 13:49:11 INFO Completed PreFailoverProcesses hook 1 of 1 in 2.708585ms
2018-05-23 13:49:11 INFO topology_recovery: done running PreFailoverProcesses hooks
2018-05-23 13:49:11 INFO topology_recovery: RecoverDeadMaster: will recover ubuntu:3306
2018-05-23 13:49:11 INFO topology_recovery: RecoverDeadMaster: masterRecoveryType=MasterRecoveryGTID
2018-05-23 13:49:11 INFO topology_recovery: RecoverDeadMaster: regrouping replicas via GTID
2018-05-23 13:49:11 DEBUG Stopping 1 replicas via StopReplicationNicely
2018-05-23 13:49:11 INFO Stopped slave nicely on ubuntu2:3306, Self:log-bin-local.000001:154, Exec:log-bin-local.000001:154
2018-05-23 13:49:11 INFO Stopped slave on ubuntu2:3306, Self:log-bin-local.000001:154, Exec:log-bin-local.000001:154
2018-05-23 13:49:11 DEBUG - sorted replica: ubuntu2:3306 log-bin-local.000001:154
2018-05-23 13:49:11 DEBUG GetCandidateReplica: candidate: ubuntu2:3306, ahead: 0, equal: 0, late: 0, break: 0
2018-05-23 13:49:11 DEBUG RegroupReplicasGTID: working on 0 replicas
2018-05-23 13:49:11 INFO Started slave on ubuntu2:3306
2018-05-23 13:49:11 DEBUG RegroupReplicasGTID: done
2018-05-23 13:49:11 INFO auditType:regroup-replicas-gtid instance:ubuntu:3306 cluster:ubuntu:3306 message:regrouped replicas of ubuntu:3306 via GTID; promoted ubuntu2:3306
2018-05-23 13:49:11 INFO auditType:begin-downtime instance:ubuntu:3306 cluster:ubuntu:3306 message:owner: ubuntu3, reason: lost-in-recovery
2018-05-23 13:49:11 INFO topology_recovery: RecoverDeadMaster: 0 postponed functions
2018-05-23 13:49:11 INFO topology_recovery: checking if should replace promoted replica with a better candidate
2018-05-23 13:49:11 INFO topology_recovery: + sanity check: found our very own server to promote; doing nothing
2018-05-23 13:49:11 INFO topology_recovery: promoted instance ubuntu2:3306 requires no further action
2018-05-23 13:49:11 INFO topology_recovery: promoted replica: ubuntu2:3306
2018-05-23 13:49:11 INFO auditType:recover-dead-master instance:ubuntu:3306 cluster:ubuntu:3306 message:promoted replica: ubuntu2:3306
2018-05-23 13:49:11 INFO topology_recovery: RecoverDeadMaster: successfully promoted ubuntu2:3306
2018-05-23 13:49:11 INFO topology_recovery: - RecoverDeadMaster: promoted server coordinates: log-bin-local.000001:154
2018-05-23 13:49:11 INFO topology_recovery: - RecoverDeadMaster: will apply MySQL changes to promoted master
2018-05-23 13:49:11 INFO Will reset replica on ubuntu2:3306
2018-05-23 13:49:11 INFO auditType:begin-maintenance instance:ubuntu2:3306 cluster:ubuntu:3306 message:maintenanceToken: 42, owner: ubuntu3, reason: reset replica
2018-05-23 13:49:11 INFO Stopped slave on ubuntu2:3306, Self:log-bin-local.000001:154, Exec:log-bin-local.000001:154
2018-05-23 13:49:12 INFO Reset slave ubuntu2:3306
2018-05-23 13:49:12 INFO auditType:reset-slave instance:ubuntu2:3306 cluster:ubuntu:3306 message:ubuntu2:3306 replication reset
2018-05-23 13:49:12 INFO auditType:end-maintenance instance:ubuntu2:3306 cluster:ubuntu:3306 message:maintenanceToken: 42
2018-05-23 13:49:12 INFO instance ubuntu2:3306 read_only: false
2018-05-23 13:49:12 INFO auditType:read-only instance:ubuntu2:3306 cluster:ubuntu:3306 message:set as false
2018-05-23 13:49:12 INFO topology_recovery: - RecoverDeadMaster: have applied read-only=0, RESET SLAVE ALL on promtoed master
2018-05-23 13:49:12 INFO topology_recovery: Writing KV [mysql/master/ashburn_cluster:ubuntu2:3306 mysql/master/ashburn_cluster/hostname:ubuntu2 mysql/master/ashburn_cluster/port:3306 mysql/master/ashburn_cluster/ipv4:192.168.0.38 mysql/master/ashburn_cluster/ipv6:]
2018-05-23 13:49:12 INFO topology_recovery: Running 1 PostMasterFailoverProcesses hooks
2018-05-23 13:49:12 INFO topology_recovery: Running PostMasterFailoverProcesses hook 1 of 1: echo 'Recovered from DeadMaster on ubuntu:3306. Failed: ubuntu:3306; Promoted: ubuntu2:3306' >> /tmp/recovery.log
2018-05-23 13:49:12 ERROR ReadTopologyInstance(ubuntu:3306) show slave hosts: ReadTopologyInstance(ubuntu:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-23 13:49:12 INFO CommandRun(echo 'Recovered from DeadMaster on ubuntu:3306. Failed: ubuntu:3306; Promoted: ubuntu2:3306' >> /tmp/recovery.log,[])
2018-05-23 13:49:12 INFO CommandRun/running: bash /tmp/orchestrator-process-cmd-546597151
2018-05-23 13:49:12 INFO CommandRun successful. exit status 0
2018-05-23 13:49:12 INFO topology_recovery: Completed PostMasterFailoverProcesses hook 1 of 1 in 2.021593ms
2018-05-23 13:49:12 INFO Completed PostMasterFailoverProcesses hook 1 of 1 in 2.021593ms
2018-05-23 13:49:12 INFO topology_recovery: done running PostMasterFailoverProcesses hooks
2018-05-23 13:49:12 INFO topology_recovery: - RecoverDeadMaster: updating cluster_alias: ubuntu:3306 -> ubuntu2:3306
2018-05-23 13:49:12 INFO Topology recovery: {"Id":52,"UID":"1527097751672896241:ae9de379c4869948a7f24c4e465ff8175e1e1c795fee9d0377ed7169819f3022","AnalysisEntry":{"AnalyzedInstanceKey":{"Hostname":"ubuntu","Port":3306},"AnalyzedInstanceMasterKey":{"Hostname":"","Port":0},"ClusterDetails":{"ClusterName":"ubuntu:3306","ClusterAlias":"ashburn_cluster","ClusterDomain":"","CountInstances":2,"HeuristicLag":0,"HasAutomatedMasterRecovery":true,"HasAutomatedIntermediateMasterRecovery":false},"AnalyzedInstanceDataCenter":"","AnalyzedInstancePhysicalEnvironment":"","IsMaster":true,"IsCoMaster":false,"LastCheckValid":true,"LastCheckPartialSuccess":true,"CountReplicas":1,"CountValidReplicas":1,"CountValidReplicatingReplicas":0,"CountReplicasFailingToConnectToMaster":0,"CountStaleReplicas":0,"CountDowntimedReplicas":0,"ReplicationDepth":0,"SlaveHosts":[{"Hostname":"ubuntu2","Port":3306}],"IsFailingToConnectToMaster":false,"Analysis":"DeadMaster","Description":"Master is reachable but its single slave is not replicating","StructureAnalysis":null,"IsDowntimed":false,"IsReplicasDowntimed":false,"DowntimeEndTimestamp":"","DowntimeRemainingSeconds":0,"IsBinlogServer":false,"PseudoGTIDImmediateTopology":false,"OracleGTIDImmediateTopology":true,"MariaDBGTIDImmediateTopology":false,"BinlogServerImmediateTopology":false,"CountStatementBasedLoggingReplicas":0,"CountMixedBasedLoggingReplicas":0,"CountRowBasedLoggingReplicas":1,"CountDistinctMajorVersionsLoggingReplicas":1,"IsActionableRecovery":true,"ProcessingNodeHostname":"ubuntu3","ProcessingNodeToken":"e212f0b44ccf54f35fef7569d499b7b231b9d9b23342aa2ee8b6b78beccb03e6","CountAdditionalAgreeingNodes":0,"StartActivePeriod":"","SkippableDueToDowntime":false,"GTIDMode":"ON","MinReplicaGTIDMode":"ON","MaxReplicaGTIDMode":"ON","CommandHint":"graceful-master-takeover"},"SuccessorKey":{"Hostname":"ubuntu2","Port":3306},"SuccessorAlias":"","IsActive":false,"IsSuccessful":true,"LostReplicas":[],"ParticipatingInstanceKeys":[],"AllErrors":[],"RecoveryStartTimestamp":"","RecoveryEndTimestamp":"","ProcessingNodeHostname":"","ProcessingNodeToken":"","Acknowledged":false,"AcknowledgedAt":"","AcknowledgedBy":"","AcknowledgedComment":"","LastDetectionId":0,"RelatedRecoveryId":0,"RecoveryType":"MasterRecoveryGTID"}
2018-05-23 13:49:12 INFO topology_recovery: Running 1 PostFailoverProcesses hooks
2018-05-23 13:49:12 INFO topology_recovery: Running PostFailoverProcesses hook 1 of 1: echo '(for all types) Recovered from DeadMaster on ubuntu:3306. Failed: ubuntu:3306; Successor: ubuntu2:3306' >> /tmp/recovery.log
2018-05-23 13:49:12 INFO CommandRun(echo '(for all types) Recovered from DeadMaster on ubuntu:3306. Failed: ubuntu:3306; Successor: ubuntu2:3306' >> /tmp/recovery.log,[])
2018-05-23 13:49:12 INFO CommandRun/running: bash /tmp/orchestrator-process-cmd-448058866
2018-05-23 13:49:12 INFO CommandRun successful. exit status 0
2018-05-23 13:49:12 INFO topology_recovery: Completed PostFailoverProcesses hook 1 of 1 in 3.806898ms
2018-05-23 13:49:12 INFO Completed PostFailoverProcesses hook 1 of 1 in 3.806898ms
2018-05-23 13:49:12 INFO topology_recovery: done running PostFailoverProcesses hooks
2018-05-23 13:49:12 INFO topology_recovery: Waiting for 0 postponed functions
2018-05-23 13:49:12 DEBUG PostponedFunctionsContainer: waiting on 0 postponed functions
2018-05-23 13:49:12 DEBUG PostponedFunctionsContainer: done waiting
2018-05-23 13:49:12 INFO topology_recovery: Executed 0 postponed functions
2018-05-23 13:49:12 ERROR ReadTopologyInstance(ubuntu:3306) show slave hosts: ReadTopologyInstance(ubuntu:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-23 13:49:12 INFO instance ubuntu:3306 read_only: true
2018-05-23 13:49:12 INFO auditType:read-only instance:ubuntu:3306 cluster:ubuntu:3306 message:set as true
2018-05-23 13:49:12 ERROR ReadTopologyInstance(ubuntu:3306) show slave hosts: ReadTopologyInstance(ubuntu:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-23 13:49:12 DEBUG ChangeMasterTo: will attempt changing master on ubuntu:3306 to ubuntu2:3306, log-bin-local.000001:154
2018-05-23 13:49:12 INFO ChangeMasterTo: Changed master on ubuntu:3306 to: ubuntu2:3306, log-bin-local.000001:154. GTID: true
2018-05-23 13:49:12 ERROR ReadTopologyInstance(ubuntu:3306) show slave hosts: ReadTopologyInstance(ubuntu:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-23 13:49:12 ERROR ReadTopologyInstance(ubuntu:3306) show slave hosts: ReadTopologyInstance(ubuntu:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-23 13:49:12 DEBUG ChangeMasterTo: will attempt changing master credentials on ubuntu:3306
2018-05-23 13:49:12 INFO ChangeMasterTo: Changed master credentials on ubuntu:3306
2018-05-23 13:49:12 ERROR ReadTopologyInstance(ubuntu:3306) show slave hosts: ReadTopologyInstance(ubuntu:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-23 13:49:12 INFO topology_recovery: Running 1 PostGracefulTakeoverProcesses hooks
2018-05-23 13:49:12 INFO topology_recovery: Running PostGracefulTakeoverProcesses hook 1 of 1: echo 'Planned takeover complete' >> /tmp/recovery.log
2018-05-23 13:49:12 INFO CommandRun(echo 'Planned takeover complete' >> /tmp/recovery.log,[])
2018-05-23 13:49:12 INFO CommandRun/running: bash /tmp/orchestrator-process-cmd-680785577
2018-05-23 13:49:12 INFO CommandRun successful. exit status 0
2018-05-23 13:49:12 INFO topology_recovery: Completed PostGracefulTakeoverProcesses hook 1 of 1 in 23.543732ms
2018-05-23 13:49:12 INFO Completed PostGracefulTakeoverProcesses hook 1 of 1 in 23.543732ms
2018-05-23 13:49:12 INFO topology_recovery: done running PostGracefulTakeoverProcesses hooks
[martini] Completed 200 OK in 1.083063479s
[martini] Started GET /web/cluster/alias/ashburn_cluster for 127.0.0.1:39674
[martini] Completed 200 OK in 5.659519ms
[martini] Started GET /api/clusters-info for 127.0.0.1:39674
[martini] Completed 200 OK in 1.491596ms
[martini] Started GET /api/cluster/ubuntu2:3306 for 127.0.0.1:39674
[martini] Started GET /api/cluster-info/ubuntu2:3306 for 127.0.0.1:39668
[martini] Completed 200 OK in 1.871358ms
[martini] Completed 200 OK in 10.724443ms
[martini] Started GET /api/active-cluster-recovery/ubuntu2:3306 for 127.0.0.1:39668
[martini] Started GET /api/recently-active-cluster-recovery/ubuntu2:3306 for 127.0.0.1:39674
[martini] Started GET /api/blocked-recoveries/cluster/ubuntu2:3306 for 127.0.0.1:39670
[martini] Completed 200 OK in 2.691813ms
[martini] Completed 200 OK in 3.797475ms
[martini] Completed 200 OK in 5.524563ms
[martini] Started GET /api/problems/ubuntu2:3306 for 127.0.0.1:39674
[martini] Completed 200 OK in 2.846492ms
[martini] Started GET /api/audit-failure-detection/alias/ashburn_cluster for 127.0.0.1:39668
[martini] Started GET /api/audit-recovery/alias/ashburn_cluster for 127.0.0.1:39674
[martini] Completed 200 OK in 19.354661ms
[martini] Completed 200 OK in 22.843289ms
[martini] Started GET /api/maintenance for 127.0.0.1:39668
[martini] Started GET /api/replication-analysis/ubuntu2:3306 for 127.0.0.1:39674
[martini] Completed 200 OK in 8.53197ms
[martini] Completed 200 OK in 14.513787ms
[martini] Started GET /api/maintenance for 127.0.0.1:39674
[martini] Completed 200 OK in 1.915259ms
[martini] Started GET /api/recently-active-instance-recovery/ubuntu2/3306 for 127.0.0.1:39674
[martini] Completed 200 OK in 5.292649ms
2018-05-23 13:49:13 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu2:3306
2018-05-23 13:49:14 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu2:3306
2018-05-23 13:49:15 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu2:3306
2018-05-23 13:49:16 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu2:3306
2018-05-23 13:49:17 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu2:3306
2018-05-23 13:49:18 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu2:3306
2018-05-23 13:49:18 DEBUG outdated keys: [ubuntu:3306 ubuntu2:3306]
2018-05-23 13:49:18 ERROR ReadTopologyInstance(ubuntu:3306) show slave hosts: ReadTopologyInstance(ubuntu:3306) 'show slave hosts' returned row with <host,port>: <,3306>
2018-05-23 13:49:18 DEBUG Discovered host: ubuntu:3306, master: ubuntu2:3306, version: 5.7.17-13-log in 0.014s (Backend: 0.006s, Instance: 0.009s)
2018-05-23 13:49:18 DEBUG Discovered host: ubuntu2:3306, master: :0, version: 5.7.17-13-log in 0.014s (Backend: 0.004s, Instance: 0.012s)
2018-05-23 13:49:19 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu2:3306
2018-05-23 13:49:20 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu2:3306
2018-05-23 13:49:21 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu2:3306
2018-05-23 13:49:22 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu2:3306
2018-05-23 13:49:23 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu2:3306
2018-05-23 13:49:24 WARNING executeCheckAndRecoverFunction: ignoring analysisEntry that has no action plan: MasterSingleSlaveNotReplicating; key: ubuntu2:3306
20


![orchestrator](https://user-images.githubusercontent.com/16723860/40443499-8152eb04-5e94-11e8-903c-0d7087ecfeec.png)

**Comment by shlomi-noach** [comment]
> The only think is that I have to issue a Start Slave command from the web UI, and then the graceful takeover is successful.

That is actually intentional. A graceful failover connects the old master back but does not start replication.

**Comment by shlomi-noach** [comment]
Elaborating a bit, and this should go into the docs. There are various scenarios to a graceful-takeover.

- In one scenario `orchestrator` wouldn't actually have access to replication credentials, so couldn't start replication if it wanted.
- In another scenario you will have upgraded your topology from `5.7` to `8.0` and the old master (`5.7`) shouldn't replicate from the new master (`8.0`). You need to upgrade.
- And if `orchestrator` does have credentials, and if versions/configuration _are_ compatible, there may be an operational reason for you to not want to start replication on the old master. There may have been a reason for demoting it and you do not want to contaminate evidence. 

Agreed or disagreed, I'm condifent that if we did configure it to auto-start, a user would point out this is wrong.

So a few options:

Run this through hooks:
```
  "PostGracefulTakeoverProcesses": [
    "orchestrator-client -c start-slave -i {failedHost}:{failedPort}"
  ],
```

Or, argue that this feature is super important, and maybe we introduce a configuration variable to auto-start (which I don't like) or another method of telling a graceful-takeover to auto start replication on old master.

**Comment by robertmyhill3** [comment]
Thanks for your thoughts and explanations.

Personally I favor adding a configuration variable to auto-restart, as that would seem to make everybody happy.  It could default to the current behavior.

Thanks again.

**Comment by robertmyhill3** [comment]
I tried adding the PostGracefulTakeoverProcess hook, as you suggested.  I'm getting the following error (see below).

In browsing the code, it looks like apiSynonyms has a start-slave->start-replica mapping, but I don't see an implementation of start-replica.  I'm just starting to learn the code, but it looks like we need a CLI implementation for start-replica?

++++++++++++++++++++++++++++++++++++++++++++++++

2018-05-24 10:42:17 INFO topology_recovery: Running PostGracefulTakeoverProcesses hook 2 of 3: orchestrator-client -c slart-slave -i ubuntu2:3306
2018-05-24 10:42:17 INFO CommandRun(orchestrator-client -c slart-slave -i ubuntu2:3306,[])
2018-05-24 10:42:17 INFO CommandRun/running: bash /tmp/orchestrator-process-cmd-819204393
2018-05-24 10:42:17 INFO CommandRun/stderr: orchestrator-client[20160]: Unsupported command slart-replica
2018-05-24 10:42:17 ERROR CommandRun: failed. exit status 1
2018-05-24 10:42:17 ERROR exit status 1
2018-05-24 10:42:17 INFO topology_recovery: Execution of PostGracefulTakeoverProcesses hook 2 of 3 failed in 25.607915ms with error: exit status 1
2018-05-24 10:42:17 ERROR Execution of PostGracefulTakeoverProcesses hook 2 of 3 failed in 25.607915ms with error: exit status 1

**Comment by shlomi-noach** [comment]
“slart-replica” probably my typo in code. 
Use “start-replica” instead of “start-slave”?

**Comment by shlomi-noach** [comment]
Or maybe it’s your typo 😛

**Comment by shlomi-noach** [comment]
Or then again, mine? Will look into it in a couple days though.

**Comment by robertmyhill3** [comment]
My apologies, that was definitely my typo :-(

But after fixing the typo, it still looks like the new slave is not getting started.

Here's what I'm seeing in the log...

2018-05-24 14:01:17 INFO topology_recovery: Running PostGracefulTakeoverProcesses hook 2 of 3: orchestrator-client -c start-slave -i ubuntu2:3306
2018-05-24 14:01:17 INFO CommandRun(orchestrator-client -c start-slave -i ubuntu2:3306,[])
2018-05-24 14:01:17 INFO CommandRun/running: bash /tmp/orchestrator-process-cmd-352388608
2018-05-24 14:01:17 INFO CommandRun successful. exit status 0
2018-05-24 14:01:17 INFO topology_recovery: Completed PostGracefulTakeoverProcesses hook 2 of 3 in 59.800146ms
2018-05-24 14:01:17 INFO Completed PostGracefulTakeoverProcesses hook 2 of 3 in 59.800146ms

...that looks good, but the new slave is not getting started.

Additionally, if I do this...

orchestrator-client -c start-replica -i ubuntu2:3306

...then the new slave is still not getting started.

But if I do this...

MYSQL_PWD='<password>' mysql --protocol=TCP -h'ubuntu2' -u<user> -e "START SLAVE";

...then the slave is successfully started.

So it looks like the "START SLAVE" is not reaching the new slave for some reason.

Any ideas on how to debug this further?

Best,
Robert

**Comment by shlomi-noach** [comment]
`start slave/stop slave` statements are logged in the MySQL error log. Can you `tail -f /path/to/your/mysql.err` and see what happens when you issue `orchestrator-client -c start-replica -i ubuntu2:3306`?

Can you try via the web UI and see if `start slave` or `stop slave` operations are successful? (You will need to click the config dialog icon for `ubuntu2:3306`).

Does `orchestrator` have `REPLICATION CLIENT` or `SUPER` privileges on your topology servers?

**Comment by robertmyhill3** [comment]
Hi there,

Thanks again for your help.

Below is (hopefully) everything you've asked for.

Does this shed any light?

Robert


+++++++++++++++++++++++++++++++++++++
orchestrator-client -c start-replica -i ubuntu2:3306
+++++++++++++++++++++++++++++++++++++
- STOP SLAVE operation is not successful
- nothing emitted into error log

++++++++++++++++++++
STOP SLAVE from Web UI:
++++++++++++++++++++
- STOP SLAVE operation is successful
- nothing emitted into error log

++++++++++++++++++++
START SLAVE from Web UI:
++++++++++++++++++++
- START SLAVE operation is successful
- error log emits the following...

2018-05-29T15:15:55.669130Z 1782 [Warning] Storing MySQL user name or password information in the master info repository is not secure and is therefore not recommended. Please consider using the USER and PASSWORD connection options for START SLAVE; see the 'START SLAVE Syntax' in the MySQL Manual for more information.
2018-05-29T15:15:55.669734Z 1783 [Warning] Slave SQL for channel '': If a crash happens this configuration does not guarantee that the relay log info will be consistent, Error_code: 0

++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
mysql slave> show grants for orchestrator@'192.168.0.125';
+------------------------------------------------------------------------------------------+
| Grants for orchestrator@192.168.0.125                                                    |
+------------------------------------------------------------------------------------------+
| GRANT RELOAD, PROCESS, SUPER, REPLICATION SLAVE ON *.* TO 'orchestrator'@'192.168.0.125' |
| GRANT SELECT ON `mysql`.`slave_master_info` TO 'orchestrator'@'192.168.0.125'            |
+------------------------------------------------------------------------------------------+


++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
mysql master> show grants for orchestrator@'192.168.0.125';
+------------------------------------------------------------------------------------------+
| Grants for orchestrator@192.168.0.125                                                    |
+------------------------------------------------------------------------------------------+
| GRANT RELOAD, PROCESS, SUPER, REPLICATION SLAVE ON *.* TO 'orchestrator'@'192.168.0.125' |
| GRANT SELECT ON `mysql`.`slave_master_info` TO 'orchestrator'@'192.168.0.125'            |
+------------------------------------------------------------------------------------------+

**Comment by robertmyhill3** [comment]
Please looks at the second of the two posts I just made, and ignore the first of the two posts.

It looks like a had some control characters in the first post that caused some things not to appear in the post.

**Comment by shlomi-noach** [comment]
deleted the first to avoid confusion.

**Comment by shlomi-noach** [comment]
OK, let's drill into:

```


+++++++++++++++++++++++++++++++++++++
orchestrator-client -c start-replica -i ubuntu2:3306
+++++++++++++++++++++++++++++++++++++

    STOP SLAVE operation is not successful
    nothing emitted into error log

```
first, I suppose a typo above: you issued a `start-replica` and then `STOP SLAVE operation is not successful`, you probably meant `START SLAVE operation not successful`.

It's interesting that the web interface made it but `orchestrator-client` did not: they both use the exact same web API.

Can you please run `orchestrator-client -c which-api` and see if it returns the same path as your web interface URL?

I suspect the above will give us the answer.

But while we're at it, will you please:

```shell
curl $(orchestrator-client -c which-api)/start-replica/ubuntu2/3306
```

**Comment by robertmyhill3** [comment]
Yes, sorry about that typo.  You're correct that my comment should have said "START SLAVE operation not successful"

orchestrator-client -c which-api responds with "http://localhost:3000/api"

So that's the problem.  I've been running with "ListenAddress": ":3010" in my orchestrator.conf.json file.  We have a different application that is using port 3000, which is why I changed Orchestrator to listen on port 3010.

I point my browser to localhost:3010, which is why I can successfully issue commands from the Web API.

I don't see how to point orchestrator-client at port 3010.  Is there a way to do that?

Also, if the web UI is listening on port 3010, and orchestrator-client is called from the web UI, then shouldn't the web UI call into the orchestrator-client  in such a way that orchestrator-client uses port 3010?

Best,
Robert

**Comment by shlomi-noach** [comment]
> I don't see how to point orchestrator-client at port 3010. Is there a way to do that?

Three ways to do it:

- edit `/etc/profile.d/orchestrator-client.sh` and put `ORCHESTRATOR_API=whatever:3010`
- run `orchestrator-client -api whatever:3010 -c start-replica ...`
- run `ORCHESTRATOR_API=whatever:3010 orchestrator-client -c start-replica`

see https://github.com/github/orchestrator/blob/master/docs/orchestrator-client.md

> Also, if the web UI is listening on port 3010, and orchestrator-client is called from the web UI, then shouldn't the web UI call into the orchestrator-client in such a way that orchestrator-client uses port 3010?

it's an external invocation ; I see where you're headed, it would mean magic and potentially confuse the user. I'll think about it.

**Comment by robertmyhill3** [comment]
I set /etc/profile.d/orchestrator-client.sh as you suggested, and that fixed the issue.

Thank you very much for your help.  I appreciate it.

Best,
Robert
